Notes/Notes/Note/View/NoteViewController.swift

```diff
@@ -0,0 +1,192 @@
+        updateBarButtonItems()
+    }
+    
+    @objc
+    private func deleteAction() {
+        viewModel?.delete()
+        navigationController?.popViewController(animated: true)
+        updateBarButtonItems()
+    }
+    
+    private func setupUI() {
+        view.addSubview(attachmentView)
+        view.addSubview(textView)
+        view.backgroundColor = .white
+        
+        textView.delegate = self
+        
+        let recogniser = UITapGestureRecognizer(target: self, action: #selector(hideKeyboard))
+        view.addGestureRecognizer(recogniser)
+        
+        textView.layer.borderWidth = textView.text.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty ? 1 : 0
+        
+        setupConstraints()
+        setupBars()
+        updateBarButtonItems()
+    }
+    
+    private func setupConstraints() {
+        attachmentView.snp.makeConstraints { make in
+            let height = attachmentView.image != nil ? imageHeight : 0
+            make.height.equalTo(height)
+            make.top.leading.trailing.equalTo(view.safeAreaLayoutGuide).inset(10)
+        }
+        
+        textView.snp.makeConstraints { make in
+            make.top.equalTo(attachmentView.snp.bottom).offset(10)
+            make.leading.trailing.equalToSuperview().inset(10)
+            make.bottom.equalTo(view.safeAreaLayoutGuide.snp.bottom).inset(-10)
+        }
+    }
+    
+    private func updateImageHeight() {
+        attachmentView.snp.updateConstraints { make in
+            make.height.equalTo(imageHeight)
+        }
+    }
+    
+    private func setImageHeight() {
+        
+        attachmentView.snp.makeConstraints { make in
+            
+        }
+    }
+    
+    @objc
+    private func hideKeyboard() {
+        textView.resignFirstResponder()
+    }
+    
+    @objc
+    private func addImage() {
+        let imagePicker = UIImagePickerController()
+        imagePicker.delegate = self
+        imagePicker.sourceType = .photoLibrary
+        
+        present(imagePicker, animated: true)
+    }
+    
+    private func setupBars() {
+        navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .save, target: self, action: #selector(saveAction))
+        
+        let categoryButton = UIBarButtonItem(title: "Category", style: .plain, target: self, action: #selector(selectCategoryAction))
+        
+        let photoButton = UIBarButtonItem(barButtonSystemItem: .camera, target: self, action: #selector(addImage))
+        
+        let space = UIBarButtonItem(systemItem: .flexibleSpace)
+        
+        setToolbarItems([categoryButton, space, photoButton, space], animated: true)
+    }
+    
+    
+    @objc
+    private func selectCategoryAction() {
+        print("Select Category Button Pressed")
+    }
+    
+    private func updateBarButtonItems() {
+        let saveButton = UIBarButtonItem(barButtonSystemItem: .save, target: self, action: #selector(saveAction))
```

Comment:
you need to create this button once and then just change saveButton.isEnabled when you have any changes for  a note.